account_invoice_change_currency/models/account_change_currency.py

```diff
@@ -36,12 +68,49 @@ def action_account_change_currency(self):
                 line.price_unit *= rate
             for tax in invoice.tax_line_ids:
                 tax.amount *= rate
+            invoice.date_invoice = today
+
+    @api.onchange('currency_id')
+    def onchange_currency(self):
```

Comment:
Use a non-common name since that could be duplicated and use private `_` prefix:
`def _onchange_currency_change_rate(self):`